### PR TITLE
Add DALI video plugin 

### PR DESCRIPTION
**Pull request by jantonguirao**
## Category:

**New feature**

## Description:

Starts the skeleton of an upcoming DALI video plugin, as a source distribution that depends of PyNvVideoCodec for usage of VideoCodecSDK and utilities.

### Affected modules and functionalities:

- New plugin

### Key points relevant for the review:

Approach to source distribution plugins

### Tests:

- [ ] Existing tests apply
- [x] New tests added
  - [x] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [x] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [x] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: DALI-3866

### CODE REVIEW

File: plugins/video/pkg_src/build_ffmpeg.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  --disable-encoders \
+  --disable-hwaccels \
+  --disable-muxers \
+  --disable-protocols \
+  --disable-indevs \
+  --disable-outdevs \
+  --disable-devices \
+  --disable-filters \
+  --disable-bsfs \
+  --disable-decoder=ipu
```
[the remark this answers is not in the document]
Comment (jantonguirao) [reply plugins/video/pkg_src/build_ffmpeg.sh:54]: Done

File: plugins/setup.py.in [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    skbuild.setup(
+        name="nvidia-dali-@DALI_PLUGIN_NAME@@DALI_FLAVOR_MINUS@",
+        description="@DALI_PLUGIN_DESCRIPTION@. For NVIDIA DALI, @DALI_FLAVOR@. Git SHA: @GIT_SHA@",
+        url='https://github.com/NVIDIA/DALI',
+        version='@DALI_VERSION@',
+        author='NVIDIA Corporation',
+        license='Apache License 2.0',
+        license_files = ('LICENSE', 'COPYRIGHT', 'Acknowledgements.txt'),
+        classifiers=[
```
[the remark this answers is not in the document]
Comment (jantonguirao) [reply plugins/setup.py.in:58]: done

File: docker/Dockerfile.make_sdist_plugins [hunk 3]
```diff
@@ -0,0 +1,18 @@
+ARG BASE_IMAGE=ubuntu:22.04
+FROM ${BASE_IMAGE}
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+RUN apt update && apt install -y python3 python3-pip build-essential cmake git && \
+    python3 -m pip install setuptools scikit-build && \
+    rm -rf /root/.cache/pip/
```
[the remark this answers is not in the document]
Comment (jantonguirao) [reply docker/Dockerfile.make_sdist_plugins:9]: done

**Comment by JanuszL** [comment]
Can you also adjust cmake/lint.cmake and .github/workflows/lint.yml to scan video plugin?

**Review by awolant — APPROVED** [review]
LGTM

File: dali/test/python/decoder/test_video.py [hunk 4]
```diff
@@ -64,13 +64,14 @@ def video_loader(batch_size, epochs):
         yield batch
 
 
-def video_decoder_iter(batch_size, epochs=1, device="cpu"):
+def video_decoder_iter(batch_size, epochs=1, device="cpu", module=fn.experimental):
```
Comment (mzient) [inline dali/test/python/decoder/test_video.py:67]: What is the module for, if there's only one value ever used? Do you predict that we'll have two different modules at some point?

File: dali/test/python/test_dali_video_plugin.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    def test_import_dali_video_ok(self):
+        import nvidia.dali.plugin.video as dali_video  # noqa: F401
+
+        assert True
+
+
+class TestDaliVideoPluginLoadFail(unittest.TestCase):
+    def test_import_dali_video_load_fail(self):
+        with self.assertRaises(Exception):
+            import nvidia.dali.plugin.video as dali_video  # noqa: F401
```
Comment (mzient) [inline dali/test/python/test_dali_video_plugin.py:28]: 1. Do we really need this kind of test? It seems overly defensive and doesn't bring much value in a controlled test environment.
2. If you insist on keeping it - what happens if you just run this test file? Will fail for being able to run the plugin? Can you somehow disable this test by default?

File: dali/test/python/decoder/test_video.py (hunk 4 above)
Comment (jantonguirao) [reply dali/test/python/decoder/test_video.py:67]: I was hoping that when we have the plugin with the alternative decoder, we can easily reuse this test to test that one, as they have the same API

File: dali/test/python/test_dali_video_plugin.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+# limitations under the License.
+
+import unittest
+
+
+class TestDaliVideoPluginLoadOk(unittest.TestCase):
+    def test_import_dali_video_ok(self):
+        import nvidia.dali.plugin.video as dali_video  # noqa: F401
+
+        assert True
```
[the remark this answers is not in the document]
Comment (jantonguirao) [reply dali/test/python/test_dali_video_plugin.py:22]: Not really

File: dali/test/python/test_dali_video_plugin.py (hunk 5 above)
Comment (jantonguirao) [reply dali/test/python/test_dali_video_plugin.py:28]: OK, removing the negative test

**Review by mzient — APPROVED** [review]

**Pull request merged**